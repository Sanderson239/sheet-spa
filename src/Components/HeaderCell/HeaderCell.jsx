import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';
import { connect } from 'react-redux';
import injectWidgetId from '../../utils/utils';


export const HeaderCell = ({ headerCellId, headerCellsById }) => (
  <Table.HeaderCell>{`${headerCellsById[headerCellId]}`}</Table.HeaderCell>
);


HeaderCell.propTypes = {
  headerCellId: PropTypes.number.isRequired,
  headerCellsById: PropTypes.shape({ 1: PropTypes.string }).isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const id = ownProps.widgetId;
  const headerCellIds = state.widgets.byId[id].header.headerCellIds;
  const headerCellsById = state.widgets.byId[id].header.headerCellsById;
  return {
    headerCellIds,
    headerCellsById,
  };
};

export default injectWidgetId(connect(mapStateToProps)(HeaderCell));
