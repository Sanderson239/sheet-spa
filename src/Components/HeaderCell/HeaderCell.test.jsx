import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HeaderCell } from './HeaderCell';

describe('Spreadsheet HeaderCell', () => {
  it('should render a HeaderCell with headerCellId and headerCellsById', () => {
    const headerCellId = 1;
    const headerCellsById = { 1: 'AcolumnName', 2: 'BcolumnName', 3: 'CcolumnName', 4: 'DcolumnName', 5: 'EcolumnName' };
    const component = shallow(
      <HeaderCell
        headerCellId={headerCellId}
        headerCellsById={headerCellsById}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
