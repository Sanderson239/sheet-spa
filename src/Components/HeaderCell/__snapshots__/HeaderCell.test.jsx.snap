// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spreadsheet HeaderCell should render a HeaderCell with headerCellId and headerCellsById 1`] = `
<TableHeaderCell
  as="th"
>
  AcolumnName
</TableHeaderCell>
`;
